### PR TITLE
fix: default to using TLSv1.3

### LINKED ISSUE #938: Prefer TLS 1.3 when possible
When possible, we should prefer TLS 1.3, or at least verify that the connector already has this behavior.

### PR DESCRIPTION
**Pull request by shubha-rajan**
fixes #938

### CODE REVIEW

**Review by enocom — APPROVED** [review]

File: core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java [hunk 1]
```diff
@@ -427,19 +427,22 @@ private SslData createSslData(
...
           throw new RuntimeException(
               String.format(
-                  "[%s] Unable to create a SSLContext for the Cloud SQL instance.", connectionName)
+                  "[%s] Unable to create a SSLContext for the Cloud SQL instance.",
+                  connectionName)
                   + " TLSv1.3 is not supported for your Java version and is required to connect"
                   + " using IAM authentication",
               ex);
+        } else {
+          logger.warning("TLSv1.3 is not supported for your Java version, fallback to TLSv1.2");
```
Comment (kurtisvg) [inline core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java:443]: Quick check - will this get logged too much? It's at the warning level and seems like it'll get called each refresh for each instance?

**Review by kurtisvg — APPROVED** [review]

File: core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java (hunk 1 above)
Comment (shubha-rajan) [reply core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java:443]: Maybe I should demote it to info? I think it's worth logging, but also don't want to clutter the logs.

Comment (kurtisvg) [reply core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java:443]: I think warning is the right level, just concerned about the frequency. 

One a refresh is maybe an acceptable amount.

Alternatively, we could move this into a helper function that only calls the warning on the first invocation, and maybe call it when the CloudSQLinstance or socketfactory is initialized.

Comment (shubha-rajan) [reply core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java:443]: I think once per refresh is fine.

Comment (shubha-rajan) [reply core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java:443]: if later we decide the logs are too cluttered, we can investigate moving it into a helper function.

**Pull request merged**